### PR TITLE
Aurora Anderson Ride Share

### PR DESCRIPTION
**Pull request by auroralemieux**
# Ride Share
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? |   Once I saw that Driver and Rider had instance methods that would use Trip's all_with_driver and all_with_driver methods, it became clear that the latter two needed to be class methods. I attempted to reuse methods and stay DRY as much as possible.    |
| Describe a concept that you gained more clarity on as you worked on this assignment. |  I gained a better understanding of what sorts of tests to write (not just how to write tests, but which tests to write). Now I can approach writing my edge cases with the mindset of "how could this be incorrectly implemented, and what could happen because of that?" instead of just "is an error raised when  incorrect arguments are passed in?"  |
| Describe a _nominal test_ that you wrote for this assignment. |  Trip.all has the nominal case "all elements are Trip instances", to check that it is behaving as expected and converting all csv data to Trip instances.
| Describe an _edge case test_ that you wrote for this assignment. |  For the same method, Trip.all, I tested to make sure that the first csv data line and the last, and a random middle one are imported.  |
| How do you feel you did in writing pseudocode first, then writing the tests and then the code? | I feel like it really worked for me, and helped me organize my thoughts and process. I took the description of each method from the assignment and wrote my it statements for my tests, then pseudocoded the methods themselves, then wrote the tests for them.  |

### CODE REVIEW

File: specs/driver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    end
+    it "all elements are Driver instances" do
+      all_drivers.each do |driver|
+        driver.must_be_kind_of Driver
+      end
+    end
+    it "grabs first line of data from csv" do
+      test_driver = nil
+      test_driver = all_drivers.find {|driver| driver.id == 1}
+      test_driver.must_be_kind_of Driver
```
Comment (CheezItMan) [inline specs/driver_spec.rb:61]: You might also need to include a check to make sure the driver's name matches.

**Comment by CheezItMan** [comment]
# Ride Share
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
|  **Baseline** |   |
|   Used Git Regularly	|   Good regular commits and nice commit messages.	|
|   Answer comprehension questions	|   Check, and I'm glad pseudo-code worked for you.	|
|  **Driver** |   |
|   Uses the `all` method in the `find` method | Check |
|   Has appropriate edge-case tests for each method in the class |  Check & nice use of the trip methods to test the driver methods.  You might also need to include a check for the first driver with find, to make sure the data of that driver is correct.    |
|   Created a method that uses a method from the `Trip` object to retrieve the list of trips | Check |
|   Created a method that uses the internal trips list to calculate the average rating |  Check |
|  **Rider** |   |
|   Uses the `all` method in the `find` method | Check |
|   Has appropriate edge-case tests for each method in the class |  You might want to check to make sure that `get_trips` returns the correct number of trips.  Similar edge case to the issue in `Driver` as well.  These are small nitpicks.  |
|   Created a method that uses a method from the `Trip` object to retrieve the list of trips | Check  |
|   Created a method that uses the internal trips method to retrieve the associated drivers | Check |
|  **Trip** |   |
|   Reads the CSV file in the `all` method | Check |
|   Has appropriate edge-case tests for each method in the class |  Similar testing edge cases, but otherwise good.  |
|   Created a method that uses a method from the `Driver` to retrieve the associated driver instance  |  Check  |
|   Created a method that uses a method from the `Rider` to retrieve the associated rider instance |  Check  |
|   Created a method to retrieve all trips by driver id |  Check   |
|   Created a method to retrieve all trips by rider id |    Check  |

## Summary

Nicely done, a few nitpicks with testing, but this was good!
